src/dsp_tools/commands/excel2json/properties.py

```diff
@@ -198,19 +207,24 @@ def _row2prop(df_row: pd.Series, row_num: int, excelfile: str) -> dict[str, Any]
         dict object of the property
 
     Raises:
-        UserError if there are any formal mistakes in the "gui_attributes" column
+        InputError: if there are any formal mistakes in the "gui_attributes" column
     """
-    _property = {x: df_row[x] for x in mandatory_properties}
-    # These are also mandatory but require formatting
-    _property.update(
-        {"labels": utl.get_labels(df_row=df_row), "super": [s.strip() for s in df_row["super"].split(",")]}
-    )
-    non_mandatory = {
-        "comments": utl.get_comments(df_row=df_row),
-        "gui_attributes": _get_gui_attribute(df_row=df_row, row_num=row_num, excelfile=excelfile),
+    _property = {x: df_row[x] for x in mandatory_properties} | {
+        "labels": utl.get_labels(df_row=df_row),
+        "super": [s.strip() for s in df_row["super"].split(",")],
     }
-    # These functions may return None, this is checked before the update
-    _property = utl.update_dict_if_not_value_none(additional_dict=non_mandatory, to_update_dict=_property)
+
+    gui_attrib = _get_gui_attribute(df_row=df_row, row_num=row_num)
+    match gui_attrib:
+        case dict():
+            _property["gui_attributes"] = gui_attrib
+        case InvalidExcelContentProblem():
+            msg = f"There is a problem with the excel file: '{excelfile}'\n" + gui_attrib.execute_error_protocol()
+            raise InputError(msg) from None
```

Comment:
These new classes are not actually exceptions. That is why they are called problem. This way it is easier to collect them than exceptions. It is not possible to raise a class that does not inherit from the Exceptions Base class To change this only in this case would in my point of view create a big inconsistency in the design.

I did not raise the exceptions on the gui_attrib class level because then the Problem classes have to know the excel name for a good error. I tried to keep the information in the classes as minimal as possible so that we can have "dumber" functions with less variables.

Additionally in my opinion the code why the try except is not much simpler.

But I agree the types in the return variables are quite horrible. We could discuss to do the pd.isnull check further outside. But I think that is something for another PR.

Reply:
additionally, we could either alias `dict[str, int | str | float]` or - imo even better - create a model for gui attribute

Reply:
I think ultimately, a model for a resource, list and property would be a good idea. But I think implementing the errors before is still a good idea as most of them come before the construction process is started. The reason why this is not checked before is because then the construction would have to be done twice, I did not want to do that.